### ISSUE TITLE
[v5.2.0] IllegalArgumentException: Unsupported delegate type crash when handle Action with BlikComponent

### ISSUE THREAD

**Issue by wrozwad** [Adyen/adyen-android#1498]
**Describe the bug**
I received the following crash when attempting to handle an action on the `BlikComponent`
```
IllegalArgumentException: Unsupported delegate type
```
<details>
<summary>Full crashlog</summary>

```
java.lang.IllegalArgumentException: Unsupported delegate type
    at com.adyen.checkout.await.internal.ui.view.AwaitView.initView(AwaitView.kt:56)
    at com.adyen.checkout.ui.core.AdyenComponentView.loadView(AdyenComponentView.kt:127)
    at com.adyen.checkout.ui.core.AdyenComponentView.access$loadView(AdyenComponentView.kt:48)
    at com.adyen.checkout.ui.core.AdyenComponentView$attach$1.invokeSuspend(AdyenComponentView.kt:104)
    at com.adyen.checkout.ui.core.AdyenComponentView$attach$1.invoke(Unknown Source:8)
    at com.adyen.checkout.ui.core.AdyenComponentView$attach$1.invoke(Unknown Source:4)
    at kotlinx.coroutines.flow.FlowKt__TransformKt$onEach$$inlined$unsafeTransform$1$2.emit(Emitters.kt:223)
    at kotlinx.coroutines.flow.StateFlowImpl.collect(StateFlow.kt:396)
    at kotlinx.coroutines.flow.StateFlowImpl$collect$1.invokeSuspend(Unknown Source:15)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
    at kotlinx.coroutines.EventLoop.processUnconfinedEvent(EventLoop.common.kt:68)
    at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:375)
    at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:30)
    at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable$default(Cancellable.kt:25)
    at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:110)
    at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:126)
    at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:56)
    at kotlinx.coroutines.BuildersKt.launch(Unknown Source:1)
    at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:47)
    at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source:1)
    at kotlinx.coroutines.flow.FlowKt__CollectKt.launchIn(Collect.kt:49)
    at kotlinx.coroutines.flow.FlowKt.launchIn(Unknown Source:1)
    at com.adyen.checkout.action.core.internal.ui.DefaultGenericActionDelegate.observeViewFlow(DefaultGenericActionDelegate.kt:152)
    at com.adyen.checkout.action.core.internal.ui.DefaultGenericActionDelegate.handleAction(DefaultGenericActionDelegate.kt:122)
    at com.adyen.checkout.action.core.internal.DefaultActionHandlingComponent.handleAction(DefaultActionHandlingComponent.kt:35)
    at com.adyen.checkout.blik.BlikComponent.handleAction(Unknown Source:12)
    at pl.superpharm.checkout.presentation.root.CheckoutScreenKt$BottomBarWithBlik$1$1.invokeSuspend(CheckoutScreen.kt:508)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
    at androidx.compose.ui.platform.AndroidUiDispatcher.performTrampolineDispatch(AndroidUiDispatcher.android.kt:81)
    at androidx.compose.ui.platform.AndroidUiDispatcher.access$performTrampolineDispatch(AndroidUiDispatcher.android.kt:41)
    at androidx.compose.ui.platform.AndroidUiDispatcher$dispatchCallback$1.run(AndroidUiDispatcher.android.kt:57)
    at android.os.Handler.handleCallback(Handler.java:942)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loopOnce(Looper.java:201)
    at android.os.Looper.loop(Looper.java:288)
    at android.app.ActivityThread.main(ActivityThread.java:7872)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```
</details>

**To Reproduce**
As described in the documentation [Create the configuration object](https://docs.adyen.com/online-payments/build-your-integration/additional-use-cases/advanced-flow-integration/?platform=Android&integration=Components&version=5.2.0#2-create-the-configuration-object), I do it with the following code:
```kotlin
// initialized via lateinit due to the requirement of using it in ComponentCallback<BlikComponentState>
lateinit val blikComponent: BlikComponent

blikComponent = BlikComponent.PROVIDER.get(
    ...
    componentCallback = object : ComponentCallback<BlikComponentState> {
        override fun onAction(action: Action) {
            // action is "await" type
            blikComponent.handleAction(action, activity)
        }
        ...
    },
)
```

**Expected behavior**
`BlikComponent` should works flawlessly with any kind of actions. Additionally there are missing details in the documentaion. On the iOS documentation, there is information about `AdyenActionComponent`. After several hours of searching, I found that we need to use `GenericActionComponent` on Android. However, all of this research contradicts the [v5.0.0 changelog](https://docs.adyen.com/pt/online-payments/release-notes/#releaseNote=2023-10-18-android-componentsdrop-in-5.0.0):
>For Components:
>- Payment method Components now handle actions. You no longer need a payment Component and action Components for a payment method with additional actions.
>- The GenericActionComponent that can handle all action types. You no longer need to implement separate Components for redirects and 3D Secure 2 authentication, for example.

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy S10e, Emulator
 - OS: Android 12, Android 14
 - SDK Version `5.2.0`

**Comment by OscarSpruit** [comment]
Hi @wrozwad, thank you for reporting this issue! We have a few ideas what might cause it, but we are not able to reproduce it. Could you share more code from your implementation, like how you deserialize the action object and how you attach the `BlikComponent` to the `AdyenComponentView`?

Every payment component should handle actions, so in your case `blikComponent.handleAction()` should be used. `GenericActionComponent` should not be used, it is only there for very advanced use cases and that's why it is also not documented.

**Comment by wrozwad** [comment]
@OscarSpruit We are using the Jetpack Compose `1.5.2`, but due to inner dependencies, we're stuck with Compose Material `1.3.1`, so I copied your `AdyenComponent` implementation. The simplified implementation looks like this:
```kotlin
@Composable
fun SomeBlikComponent(
    session: CheckoutSession,
    configuration: BlikConfiguration,
    onSubmit: (callback: SessionComponentCallback<BlikComponentState>, state: BlikComponentState) -> Unit,
) {
    lateinit var blikComponent: BlikComponent
    blikComponent = BlikComponent.PROVIDER.get(
        checkoutSession = session,
        paymentMethod = session.getPaymentMethod(PaymentMethodTypes.BLIK)!!,
        configuration = configuration,
        componentCallback = object : SessionComponentCallback<BlikComponentState> {
            override fun onSubmit(state: BlikComponentState): Boolean {
                onSubmit(this, state) // this goes up to the ViewModel
                return true
            }

            override fun onAction(action: Action) {
                blikComponent.handleAction(action, activity) // crash!
            }

            override fun onError(componentError: ComponentError) {
                // TODO handle error state
            }

            override fun onFinished(result: SessionPaymentResult) {
                // TODO handle finished state
            }
        },
        key = "123",
    )
    AdyenComponent(blikComponent)
}
```


```kotlin
class CheckoutViewModel(
    private val checkoutRepository,
): ViewModel() {

    lateinit var checkout: MyAppCheckout

    fun onSubmit(
        callback: SessionComponentCallback<BlikComponentState>,
        state: BlikComponentState,
    ) {
        viewModelScope.launch {
            try {
                val paymentData = PaymentComponentData.SERIALIZER
                    .serialize(state.data)
                    .toString()

                callback.onLoading(true)

                val order = checkoutRepository.createOrder(
                    id = checkout.id,
                    paymentMethodCode = "blik",
                    paymentData = paymentData,
                )

                if (order.action != null) { // order.action is a String type received directly from the API
                    callback.onAction(
                        Action.SERIALIZER.deserialize(
                            JSONObject(paymentStatus.data),
                        ),
                    )
                } else {
                    callback.onFinished(...)
                }
            } catch (e: Exception) {
                callback.onError(...)
            } finally {
                callback.onLoading(false)
            }
        }
    }
}
```

**Comment by wrozwad** [comment]
By the way, does the `GenericActionComponent` have a stable SDK API? We need it for our "very advanced use case" with the PBL method, so it's really good that it's available.

**Comment by OscarSpruit** [comment]
Thank you for the details! We are able to reproduce it now and we are working on a fix. It seems to be an issue caused by the order compose renders views.

Yes, `GenericActionComponent` has a stable API. It is even being used internally by our  drop-in solution.

**Comment by OscarSpruit** [comment]
> We need it for our "very advanced use case" with the PBL method, so it's really good that it's available.

@wrozwad can you explain your use case? We would like to help you find the best way to achieve it, wether that is the `GenericActionComponent` or another way.

**Comment by wrozwad** [comment]
@OscarSpruit We need a feature to remember the last payment method along with the specific selected bank for quick transfer "OnlineBanking_PL" in order to achieve "one-click checkout". We've noticed that we can easily create an instance of the `OnlineBankingPLPaymentMethod` class ourselves, allowing us to implement our own bank picker with an extra search feature. This also enables us to easily remember the ID of the specific issuer. The only thing we're missing is handling actions after the payment is processed, which we also want to do using the `GenericActionComponent`. Are we correct?

**Comment by OscarSpruit** [comment]
@wrozwad technically this should be possible and `GenericActionComponent` would indeed be the best way to handle the actions. Please be aware though that we cannot guarantee that your solution will keep working with our future backend and frontend releases.

In case you missed it: maybe the `OnlineBankingPLComponent` could be useful for you solution as well. By using `setViewType(IssuerListViewType.SPINNER_VIEW)` on `OnlineBankingPLConfiguration` you can change the appearance to a dropdown type of view.

**Comment by wrozwad** [comment]
I know I can use the `SPINNER_VIEW`, unfortunately, our UI department expressed some disapproval of this spinner, so we'll try a custom solution. Additionally, the `SPINNER_VIEW` does not support the "remember last bank" feature, which is crucial for us.

**Comment by OscarSpruit** [comment]
Thank you for the feedback. I'll take it back to the team, so we can consider this feature. Unfortunately, you will indeed have to go with a custom solution, for now.

**Comment by jreij** [comment]
We have released [5.3.0](https://github.com/Adyen/adyen-android/releases/tag/5.3.0) which includes a fix for this issue. Let us know if you still face any problems with it.
